src/session_impl.cpp

```diff
@@ -3784,12 +3786,14 @@ namespace aux {
 			if (pi->web_seed) continue;
 			torrent* t = p->associated_torrent().lock().get();
 			if (!t) continue;
+			// TODO: 3 peers should know whether their torrent is paused or not,
+			// instead of having to ask it over and over again
 			if (t->is_paused()) continue;
 
 			if (pi->optimistically_unchoked)
 			{
 				TORRENT_ASSERT(!p->is_choked());
-				opt_unchoke.push_back(peer_connection_handle(*i));
```

Comment:
I think the intent is that optimistically_unchoked implies the peer is unchoked so if it is true then the second condition will never be true due to the is_choked() term.